src/Microsoft.IdentityModel.Protocols.OpenIdConnect/OpenIdConnectProtocolValidator.cs

```diff
@@ -429,28 +429,22 @@ public virtual HashAlgorithm GetHashAlgorithm(string algorithm)
             {
                 switch (algorithm)
                 {
-                    case SecurityAlgorithms.SHA256:
```

Comment:
`GetHashAlgorithm` is used by `ValidateHash`, whose `algorithm` parameter corresponds to the `alg` header, which is necessarily a signature algorithm, not a digest algorithm.